src/lib/reasoners/intervalCalculus.ml

```diff
@@ -2097,9 +2115,19 @@ let optimizing_split env uf opt_split =
 
       | Sim.Core.Unbounded _ ->
         let value =
-          if to_max then Th_util.Pinfinity else Th_util.Minfinity
+          if to_max then
+            Objective.Value.Pinfinity
+          else
+            Objective.Value.Minfinity
         in
-        Some { opt_split with value }
+        (* As the problem is unbounded, we need to produce a value for the
+           objective function. In this case, we pick a value in the domain
+           of the polynomial [p]. *)
```

Comment:
Why do we need to explicitly pick a value for the objective? If my understanding is correct, this is only intended to be used during model generation; are the standard model generation mechanism not enough?

Reply:
See the comment below for the `Sim.Core.Max`.